CommonPackage.cmake

```diff
@@ -14,44 +25,181 @@ set(ENV{PKG_CONFIG_PATH}
+  # implement found_components check)
+  if((NOT ${Name}_FOUND AND NOT ${COMMON_PACKAGE_NAME}_FOUND) OR COMMON_PACKAGE_COMPONENTS)
+    set(COMMON_PACKAGE_VERSION)
+    if(COMMON_PACKAGE_ARGS)
+      list(GET COMMON_PACKAGE_ARGS 0 COMMON_PACKAGE_VERSION)
+      if(COMMON_PACKAGE_VERSION MATCHES "^[0-9.]+$") # is a version
+        set(COMMON_PACKAGE_VERSION ">=${COMMON_PACKAGE_VERSION}")
+      else()
+        set(COMMON_PACKAGE_VERSION)
+      endif()
+    endif()
+
+    if(COMMON_PACKAGE_QUIET)
       set(COMMON_PACKAGE_FIND_QUIET "QUIET")
+    else()
+      list(FIND COMMON_PACKAGE_ARGS "QUIET" COMMON_PACKAGE_QUIET_POS)
+      if(COMMON_PACKAGE_QUIET_POS EQUAL -1)
+        set(COMMON_PACKAGE_FIND_QUIET)
+      else()
+        set(COMMON_PACKAGE_FIND_QUIET "QUIET")
+      endif()
+    endif()
+
+    list(FIND COMMON_PACKAGE_ARGS "REQUIRED" COMMON_PACKAGE_REQUIRED_POS)
+    if(COMMON_PACKAGE_REQUIRED_POS EQUAL -1) # Optional find
+      find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
+      if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
+        pkg_check_modules(${Name} ${Name}${COMMON_PACKAGE_VERSION}
+          ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way
+      endif()
+    else() # required find
+      list(REMOVE_AT COMMON_PACKAGE_ARGS ${COMMON_PACKAGE_REQUIRED_POS})
+      find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
+      if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
+        pkg_check_modules(${Name} REQUIRED ${Name}${COMMON_PACKAGE_VERSION}
+          ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way (and fail if needed)
+      endif()
     endif()
   endif()
 
-  list(FIND COMMON_PACKAGE_ARGS "REQUIRED" COMMON_PACKAGE_REQUIRED_POS)
-  if(COMMON_PACKAGE_REQUIRED_POS EQUAL -1) # Optional find
-    find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
-    if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
-      pkg_check_modules(${Name} ${Name}${COMMON_PACKAGE_VERSION}
-        ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way
+  if(EXISTS ${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+    include(${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+  endif()
+
+  string(TOUPPER ${PROJECT_NAME} UPPER_PROJECT_NAME)
+  set(DEFDEP "${UPPER_PROJECT_NAME}_USE_${COMMON_PACKAGE_NAME}")
+  string(REGEX REPLACE "-" "_" DEFDEP ${DEFDEP})
+
+  if(${COMMON_PACKAGE_NAME}_FOUND)
+    set(${Name}_name ${COMMON_PACKAGE_NAME})
+    set(${Name}_FOUND TRUE)
+  elseif(${Name}_FOUND)
+    set(${Name}_name ${Name})
+    set(${COMMON_PACKAGE_NAME}_FOUND TRUE)
+  else()
+    set(${PROJECT_NAME}_FIND_PACKAGES_NOTFOUND "${${PROJECT_NAME}_FIND_PACKAGES_NOTFOUND} ${Name}")
+  endif()
+  if(${Name}_name)
+    list(APPEND FIND_PACKAGES_DEFINES ${DEFDEP})
```

Comment:
It's more a detail which is kindly resolved by calling common_package_post(), but I'll document the in/out variables